test/actions/helpers.test.js

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import assert from 'assert';
+
+import {UserTypes} from 'action_types';
+import {forceLogoutIfNecessary} from 'actions/helpers';
+import {Client4} from 'client';
+import {ClientError} from 'client/client4';
+import configureStore, {mockDispatch} from 'test/test_store';
+
+describe('Actions.Helpers', () => {
+    describe('forceLogoutIfNecessary', () => {
+        const token = 'token';
+
+        beforeEach(() => {
+            Client4.setToken(token);
+        });
+
+        it('should do nothing when passed a client error', async () => {
+            const store = await configureStore({
```

Comment:
Maybe we can move the store and dispatch initialization to the beforeEach, and handle the unique difference in the `'should do nothing when not logged in'` test case configuring again the store.

Reply:
I'm kind of on the fence about how this is done. On one hand, I like that this keeps all of the code for the test in one place, and it ensures that the tests don't depend on each other, but on the other, it is a lot of duplication in some of these cases.

I've wanted to write a more declarative way to construct the store for a while now (something like `const store = TestStore.addUser({current: true}).configure()`) that would kind of solve both problems, but last time I tried that, it got really complicated. Maybe it would work for simple cases like this though